### PR TITLE
refactor: ImagesDetailsComponent to Svelte5

### PR DESCRIPTION
**Pull request by feloy**
### What does this PR do?

- Use `$props()` and `$state()`
- Remove unused slot

### What issues does this PR fix or reference?

Part of #8212 

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/renderer/src/lib/image/ImageDetailsFiles.svelte [hunk 1]
```diff
@@ -11,16 +11,17 @@ import FilesystemLayerView from './FilesystemLayerView.svelte';
 import { type ImageFilesystemLayerUI, toImageFilesystemLayerUIs } from './imageDetailsFiles';
 import ImageDetailsFilesLayers from './ImageDetailsFilesLayers.svelte';
 
-export let imageInfo: ImageInfo | undefined;
+let { imageInfo }: { imageInfo: ImageInfo | undefined } = $props();
+
+let imageLayers: ImageFilesystemLayers | undefined = $state();
+let selectedLayer: ImageFilesystemLayerUI | undefined = $state();
```
Comment (axel7083) [inline packages/renderer/src/lib/image/ImageDetailsFiles.svelte]:
```suggestion
let imageLayers: ImageFilesystemLayers | undefined = $state(undefined);
let selectedLayer: ImageFilesystemLayerUI | undefined = $state(undefined);
```

Are we sure svelte is using `undefined` and not null?

File: packages/renderer/src/lib/image/ImageDetailsFiles.svelte [hunk 2]
```diff
@@ -11,16 +11,17 @@ import FilesystemLayerView from './FilesystemLayerView.svelte';
 import { type ImageFilesystemLayerUI, toImageFilesystemLayerUIs } from './imageDetailsFiles';
 import ImageDetailsFilesLayers from './ImageDetailsFilesLayers.svelte';
 
-export let imageInfo: ImageInfo | undefined;
+let { imageInfo }: { imageInfo: ImageInfo | undefined } = $props();
+
+let imageLayers = $state<ImageFilesystemLayers>();
```
Comment (axel7083) [inline packages/renderer/src/lib/image/ImageDetailsFiles.svelte:16]: What is the type of imageLayers now ?

Comment (feloy) [reply packages/renderer/src/lib/image/ImageDetailsFiles.svelte:16]: It seems to be a supported notation (and the one used in other components of podman desktop). Here, `$state<ImageFilesystemLayers>()` returns a value of type `ImageFilesystemLayers | undefined`

**Review by axel7083 — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

File: packages/renderer/src/lib/image/ImageDetailsFiles.svelte [hunk 3]
```diff
@@ -65,9 +66,6 @@ onDestroy(() => {
   </div>
 {:else if imageLayers}
   <div class="flex flex-col w-full h-full p-8 pr-0 text-[var(--pd-content-text)] bg-[var(--pd-content-bg)]">
-    <div class="pr-4">
-      <slot name="header-info" />
-    </div>
```
Comment (benoitf) [inline packages/renderer/src/lib/image/ImageDetailsFiles.svelte:70]: how is it replaced now ?

File: packages/renderer/src/lib/image/ImageDetailsFiles.svelte (hunk 2 above)
Comment (jeffmaury) [inline packages/renderer/src/lib/image/ImageDetailsFiles.svelte:16]: $state is used to declare reactive data. Is this required here ?

File: packages/renderer/src/lib/image/ImageDetailsFiles.svelte [hunk 4]
```diff
@@ -65,9 +66,6 @@ onDestroy(() => {
   </div>
 {:else if imageLayers}
   <div class="flex flex-col w-full h-full p-8 pr-0 text-[var(--pd-content-text)] bg-[var(--pd-content-bg)]">
-    <div class="pr-4">
```
Comment (jeffmaury) [inline packages/renderer/src/lib/image/ImageDetailsFiles.svelte:68]: Same concern here

File: packages/renderer/src/lib/image/ImageDetailsFiles.svelte (hunk 3 above)
Comment (feloy) [reply packages/renderer/src/lib/image/ImageDetailsFiles.svelte:70]: I removed this part as it is never used: this component is inspired from the `ProviderResultPage` component, where this slot is used, but is never used in `ImageDetailsFiles`. 

Also, moving to svelte5, the checks refuses to use this svelte4 notation, I had to change or remove it.

File: packages/renderer/src/lib/image/ImageDetailsFiles.svelte (hunk 2 above)
Comment (feloy) [reply packages/renderer/src/lib/image/ImageDetailsFiles.svelte:16]: It needs to be a state, as it is modified in line 40 (`imageLayers = layers`) and read in line 67 (`{:else if imageLayers}`)

Comment (jeffmaury) [reply packages/renderer/src/lib/image/ImageDetailsFiles.svelte:16]: Yes but it is modified in onMount so do we really need reactive here ?

Comment (feloy) [reply packages/renderer/src/lib/image/ImageDetailsFiles.svelte:16]: It is on a `subscribe` (line 31) initiated in onMount, it can happen at any time

**Review by jeffmaury — APPROVED** [review]
LGTM

**Comment by simonrey1** [comment]
@feloy for file `packages/renderer/src/lib/image/ImageDetailsFiles.svelte` I think you also need to change:

`imageLayers = await window.imageGetFilesystemLayers(provider.id, img, cancellableTokenId);`

to 

`imageLayers = await window.imageGetFilesystemLayers(provider.id, $state.snapshot(img), cancellableTokenId);`

**Pull request merged**